### PR TITLE
1st try commit 26/12

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,16 +19,33 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(word -> word.split(","))
+                .flatMap(Arrays::stream)
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:24]: You can do it in one line
```suggestion
                .flatMap(word -> Stream.of(word.split(",")))
```

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -14,16 +19,33 @@ public class StreamPractice {
...
+     * But before that subtract 1 from each element on an odd position (having the odd index)
+     * Но перед этим вычтите 1 из каждого элемента в нечетной позиции (имеющего нечетный индекс)
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(index -> index % 2 != 0
+                        ? numbers.get(index) - 1
+                        : numbers.set(index, numbers.get(index)))
+                .boxed()
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:44]:
```suggestion
```

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -14,16 +19,33 @@ public class StreamPractice {
...
+                .filter(value -> value % 2 != 0)
+                .mapToInt(Integer::valueOf)
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:46]:
```suggestion
```

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -59,7 +92,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex().equals(Person.Sex.WOMAN)
+                        && person.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:99]: You can do it in one line
```suggestion
                .flatMap(person -> person.getCats().stream())
```

Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:99]: You can do it in one line
```suggestion
                .flatMap(person -> person.getCats().stream())
```

File: src/main/java/practice/CandidateValidator.java [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MINIMAL_AGE = 35;
+    private static final String ALLOWED_NATIONALITY = "Ukrainian";
+    private static final int ALLOWED_PERIOD = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] periods = candidate.getPeriodsInUkr().split("-");
+        int fromYear = Integer.parseInt(periods[0]);
```
Comment (aleksandr-bevza) [inline src/main/java/practice/CandidateValidator.java:14]:
```suggestion
        int fromYear = Integer.parseInt(periods[PERIODS_IN_UKR_FROM_INDEX]);
```

File: src/main/java/practice/CandidateValidator.java [hunk 6]
```diff
@@ -1,5 +1,22 @@
...
+        int toYear = Integer.parseInt(periods[1]);
```
Comment (aleksandr-bevza) [inline src/main/java/practice/CandidateValidator.java:15]:
```suggestion
        int toYear = Integer.parseInt(periods[PERIODS_IN_UKR_TO_INDEX]);
```

**Review by aleksandr-bevza — CHANGES_REQUESTED** [review]

File: src/main/java/practice/CandidateValidator.java [hunk 7]
```diff
@@ -1,5 +1,24 @@
...
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MINIMAL_AGE = 35;
+    private static final String ALLOWED_NATIONALITY = "Ukrainian";
+    private static final int ALLOWED_PERIOD = 10;
```
Comment (okuzan) [inline src/main/java/practice/CandidateValidator.java:9]: MIN_ALLOWED_PERIOD conveys purpose better

File: src/main/java/practice/CandidateValidator.java [hunk 8]
```diff
@@ -1,5 +1,24 @@
...
+    private static final int PERIODS_IN_UKR_FROM_INDEX = 0;
+    private static final int PERIODS_IN_UKR_TO_INDEX = 1;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] periods = candidate.getPeriodsInUkr().split("-");
```
Comment (okuzan) [inline src/main/java/practice/CandidateValidator.java:15]: delimiter could also be constant

**Review by okuzan — APPROVED** [review]
Good job

**Pull request closed without merging**